src/license_manager_cli_ops.py

```diff
@@ -0,0 +1,202 @@
+
+        # Delete cache dir if it already exists
+        if self._CACHE_DIR.exists():
+            rmtree(self._CACHE_DIR, ignore_errors=True)
+        # Create a clean cache dir
+        self._CACHE_DIR.mkdir(parents=True)
+        chown(self._CACHE_DIR.as_posix(), self._SLURM_USER, self._SLURM_GROUP)
+        self._CACHE_DIR.chmod(0o700)
+
+
+    def setup_log_dir(self):
+        """Set up log dir."""
+
+        # Create log dir
+        if not self._LOG_DIR.exists():
+            self._LOG_DIR.mkdir(parents=True)
+        chown(self._LOG_DIR.as_posix(), self._SLURM_USER, self._SLURM_GROUP)
+        self._LOG_DIR.chmod(0o700)
+
+
+    def upgrade(self, version: str):
+        """Upgrade license-manager-cli package to specified version."""
+
+        pip_install_cmd = [
+            self._VENV_PYTHON,
+            "-m",
+            "pip",
+            "--upgrade",
+            f"{self._PACKAGE_NAME}=={version}",
+        ]
+
+        out = subprocess.check_output(pip_install_cmd).decode().strip()
+        if "Successfully installed" not in out:
+            logger.error("Trouble upgrading license-manager-cli, please debug")
+            raise Exception("license-manager-cli not upgraded.")
+        else:
+            logger.debug("license-manager-cli upgraded")
+
+        # Clear cache dir after upgrade to avoid stale data
+        self.setup_cache_dir()
+
+
+    def configure_etc_default(self):
+        """Create the default env file with the charm's configurations."""
+        charm_config = self._charm.model.config
+        license_manager_backend_url = charm_config.get("license-manager-backend-url")
+        oidc_domain = charm_config.get("oidc-domain")
+        oidc_audience = charm_config.get("oidc-audience")
+        oidc_client_id = charm_config.get("oidc-client-id")
+
+        ctxt = {
+            "license_manager_backend_url": license_manager_backend_url,
+            "oidc_domain": oidc_domain,
+            "oidc_audience": oidc_audience,
+            "oidc_client_id": oidc_client_id,
+        }
+
+        template_dir = Path("./src/templates/")
+        template_file = "license-manager-cli.defaults.template"
+        environment = Environment(loader=FileSystemLoader(template_dir))
+        template = environment.get_template(template_file)
+
+        rendered_template = template.render(ctxt)
+
+        if self._ETC_DEFAULT.exists():
+            self._ETC_DEFAULT.unlink()
+
+        self._ETC_DEFAULT.write_text(rendered_template)
+
+
+    def configure_bin_script(self):
+        """Create the lm-cli executable script."""
+
+        ctxt = {
+            "lm_cli_venv_path": self._VENV_DIR.as_posix(),
+        }
+
+        template_dir = Path("./src/templates/")
+        template_file = "license-manager-cli.script.template"
+        environment = Environment(loader=FileSystemLoader(template_dir))
+        template = environment.get_template(template_file)
+
+        rendered_template = template.render(ctxt)
```

Comment:
since it's just one argument, you could do directly:

```python
rendered_template = template.render(lm_cli_venv_path=self._VENV_DIR.as_posix())
```

Reply:
Totally, I'll change it.